tests/agent_server/test_server_agent.py

```diff
@@ -3887,3 +3899,211 @@ def test_deploy_response_matrix_invariants():
     for condition, reaction in deploy_response_matrix.items():
         if (reaction in [DeployRequestAction.interrupt, DeployRequestAction.terminate]) and condition[0][0] and condition[0][1]:
             assert not condition[1][1], "Invariant violation, regression on #6202"
+
+
+async def test_logging_failure_when_creating_venv(
+    resource_container,
+    agent: Agent,
+    client: Client,
+    clienthelper: ClientHelper,
+    environment: uuid.UUID,
+    no_agent_backoff: None,
+    caplog,
+    monkeypatch,
+):
+    """
+    Test goal: make sure that failed resources are correctly logged by the `resource_action` logger.
+    """
+
+    caplog.set_level(logging.INFO)
+    resource_container.Provider.reset()
+    agent_name = "agent1"
+    myagent_instance = agent._instances[agent_name]
+
+    resource_container.Provider.set("agent1", "key1", "value1")
+
+    def get_resources(version, value_resource_three):
+        return [
+            {
+                "key": "key1",
+                "value": "value2",
+                "id": "test::Resource[agent1,key=key1],v=%d" % version,
+                "send_event": False,
+                "purged": False,
+                "requires": [],
+            },
+            {
+                "key": "key2",
+                "value": "value2",
+                "id": "test::AgentConfig[agent1,key=key2],v=%d" % version,
+                "send_event": False,
+                "purged": False,
+                "requires": ["test::Resource[agent1,key=key1],v=%d" % version],
+            },
+            {
+                "key": "key3",
+                "value": value_resource_three,
+                "id": "test::Resource[agent1,key=key3],v=%d" % version,
+                "send_event": False,
+                "purged": False,
+                "requires": ["test::AgentConfig[agent1,key=key2],v=%d" % version],
+            },
+        ]
+
+    version1 = await clienthelper.get_version()
+
+    # Initial deploy
+    await _deploy_resources(client, environment, get_resources(version1, "value2"), version1, push=False)
+
+    async def ensure_code(code: Collection[ResourceInstallSpec]) -> executor.FailedResources:
+        raise RuntimeError(f"Failed to install handler `test` version={version1}")
+
+    monkeypatch.setattr(myagent_instance.executor_manager.process, "ensure_code", ensure_code)
+
+    await myagent_instance.get_latest_version_for_agent(
+        DeployRequest(reason="Deploy 1", is_full_deploy=False, is_periodic=False)
+    )
+
+    monkeypatch.undo()
+
+    idx1 = log_index(
+        caplog,
+        f"resource_action_logger.{environment}",
+        logging.ERROR,
+        "All resources `test::Resource` failed to load handler code or install handler code dependencies: "
+        "`Could not set up executor for agent1: Failed to "
+        "install handler `test` version=1`.",
```

Comment:
I would expect an error message + stack trace here. Because the stack trace is mostly the thing that allows you to determine the root cause of an issue.